src/main.rs

```diff
@@ -276,6 +276,15 @@ fn main() -> Result<(), Box<dyn std::error::Error>> {
         state.niri.config_error_notification.show_created(path);
     }
 
+    // Set SCHED_RESET_ON_FORK and request minimal realtime round-robin prio for main pid
+    let param = libc::sched_param {
+        sched_priority: unsafe { libc::sched_get_priority_min(libc::SCHED_RR) }
+    };
+    ;
+    if (unsafe { libc::sched_setscheduler(0, libc::SCHED_RR | libc::SCHED_RESET_ON_FORK, &param) }) {
```

Comment:
this should be changed to pthread_setschedparam like in https://github.com/KDE/kwin/commit/5921be95d3bfcf239436681bc5fe7837c5cee203 (just a bit more code). just because musl refuses to make sched_setscheduler do anything...

Reply:
also like you found in the sway discussions, this doesn't work on freebsd, and the other way with an atfork handler is more annoying...

Reply:
Eh, this was just me hoping to get by without supporting FreeBSD/musl… I'll change it.
Do we want to support setting the real-time scheduling policy on BSDs? AFAIR from my research for sway you need to write your own kernel module to support this, IMO its easier to put this behind a system feature check.

Reply:
the pthread_ version for musl is a tiny change so may as well (so i thought to mention it, not that hard)

for freebsd, no idea personally. since it seems to not be well explored (i.e. nobody actually tried to make it work on freebsd in the sway threads, there's no rtkit for things like mutter, kwin just conditionalises on the RESET_ON_FORK that doesn't exist there..) i think only requesting this on linux is ok (but leave a comment imo).

Reply:
the functionality looks correct. not sure i see the point in matching the errno code manually and printing a message, but i have no opinion really, and it's not my project :)

the only thing i'm curious about is if you notice some difference with it set. i remember it was noticeable on sway back in the day...

the biggest thing i run into sometimes is that if i run two builds or something (so load goes to 2x nthreads) then libinput drops some input events (as you mentioned up top), maybe that gets fixed :) it's super obvious too, since it's like the mouse goes to like \~10% of it's sensitivity and is really slow (since most events get dropped)